app/src/main/java/org/mozilla/fenix/library/history/HistorySearchDialogFragment.kt

```diff
@@ -0,0 +1,261 @@
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.toolbar.ToolbarPosition
+import org.mozilla.fenix.databinding.FragmentHistorySearchDialogBinding
+import org.mozilla.fenix.databinding.SearchSuggestionsHintBinding
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.library.history.awesomebar.AwesomeBarView
+import org.mozilla.fenix.library.history.toolbar.ToolbarView
+import org.mozilla.fenix.settings.SupportUtils
+
+@Suppress("TooManyFunctions")
+class HistorySearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
+    private var _binding: FragmentHistorySearchDialogBinding? = null
+    private val binding get() = _binding!!
+
+    private lateinit var interactor: HistorySearchDialogInteractor
+    private lateinit var store: HistorySearchFragmentStore
+    private lateinit var toolbarView: ToolbarView
+    private lateinit var awesomeBarView: AwesomeBarView
+
+    private var dialogHandledAction = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(STYLE_NO_TITLE, R.style.SearchDialogStyle)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        return object : Dialog(requireContext(), this.theme) {
+            override fun onBackPressed() {
+                this@HistorySearchDialogFragment.onBackPressed()
+            }
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentHistorySearchDialogBinding.inflate(inflater, container, false)
+        val activity = requireActivity() as HomeActivity
+
+        store = HistorySearchFragmentStore(
+            createInitialHistorySearchFragmentState()
+        )
+
+        interactor = HistorySearchDialogInteractor(
+            HistorySearchDialogController(
+                activity = activity,
+                fragmentStore = store,
+                clearToolbarFocus = {
+                    dialogHandledAction = true
+                    toolbarView.view.hideKeyboard()
+                    toolbarView.view.clearFocus()
+                },
+            )
+        )
+
+        toolbarView = ToolbarView(
+            context = requireContext(),
+            interactor = interactor,
+            isPrivate = false,
+            view = binding.toolbar,
+        )
+
+        val awesomeBar = binding.awesomeBar
+
+        awesomeBarView = AwesomeBarView(
+            activity,
+            interactor,
+            awesomeBar,
+        )
+
+        awesomeBarView.view.setOnEditSuggestionListener(toolbarView.view::setSearchTerms)
+        binding.searchWrapper.background = ColorDrawable(Color.TRANSPARENT)
+
+        return binding.root
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
```

Comment:
what's this suppress doing? (haven't used it before)

Reply:
Here's a good explanation I found:

"If a View that overrides onTouchEvent or uses an OnTouchListener does not also implement performClick and call it when clicks are detected, the View may not handle accessibility actions properly. Logic handling the click actions should ideally be placed in View#performClick as some accessibility services invoke performClick when a click action should occur."

I need it here is because when the user tap on anything but the toolbar, I want to close the toolbar and dismiss the keyboard.  But this code will take the user action away from the view.